build tikv failed with command "make", but succeeded with "make static_release"

It seems ROCKSDB_SYS_STATIC=1 is always needed when build.
```
// tikv/Makefile
 31 release:
 32     cargo build --release --features "${ENABLE_FEATURES}"
 33     @mkdir -p ${BIN_PATH}
 34     cp -f ${CARGO_TARGET_DIR}/release/tikv-ctl ${CARGO_TARGET_DIR}/release/tikv-server ${BIN_PATH}/
 35 
 36 static_release:
 37     ROCKSDB_SYS_STATIC=1 ROCKSDB_SYS_PORTABLE=1 make release

```

## make static_release (succeeded)
```
suzhou@suzhou-N55SL:~/zhousu/tidb/docs/scripts/deps/tikv$ make static_release
ROCKSDB_SYS_STATIC=1 ROCKSDB_SYS_PORTABLE=1 make release
make[1]: Entering directory '/home/suzhou/zhousu/tidb/docs/scripts/deps/tikv'
cargo build --release --features "default static-link portable"
   Compiling unicase v1.4.0
   Compiling thread-id v2.0.0
   Compiling cookie v0.2.5
   Compiling tikv v0.0.1 (file:///home/suzhou/zhousu/tidb/docs/scripts/deps/tikv)
   Compiling librocksdb_sys v0.1.0 (https://github.com/ngaut/rust-rocksdb.git#e2a50c47)
   Compiling thread_local v0.2.7
   Compiling backtrace v0.2.3
   Compiling regex v0.1.70
   Compiling net2 v0.2.23
   Compiling miow v0.1.2
   Compiling num v0.1.32
   Compiling chrono v0.2.25
   Compiling hyper v0.9.10
   Compiling mio v0.5.0 (https://github.com/carllerche/mio.git#7d4778f2)
   Compiling clap v2.14.0
   Compiling protobuf v1.0.23
   Compiling prometheus v0.2.8 (https://github.com/pingcap/rust-prometheus.git#b6da17ea)
   Compiling tipb v0.0.1 (https://github.com/pingcap/tipb.git#378c9b0c)
   Compiling kvproto v0.0.1 (https://github.com/pingcap/kvproto#ed926a15)
   Compiling rocksdb v0.3.0 (https://github.com/ngaut/rust-rocksdb.git#e2a50c47)
    Finished release [optimized + debuginfo] target(s) in 1217.68 secs
cp -f /home/suzhou/zhousu/tidb/docs/scripts/deps/tikv/target/release/tikv-ctl /home/suzhou/zhousu/tidb/docs/scripts/deps/tikv/target/release/tikv-server /home/suzhou/zhousu/tidb/docs/scripts/deps/tikv/bin/
make[1]: Leaving directory '/home/suzhou/zhousu/tidb/docs/scripts/deps/tikv'
```

## make (Failed)
```
suzhou@suzhou-N55SL:~/zhousu/tidb/docs/scripts/deps/tikv$ make release
cargo build --release --features "default"
   Compiling librocksdb_sys v0.1.0 (https://github.com/ngaut/rust-rocksdb.git#e2a50c47)
   Compiling tikv v0.0.1 (file:///home/suzhou/zhousu/tidb/docs/scripts/deps/tikv)
Build failed, waiting for other jobs to finish...
error: failed to run custom build command for `librocksdb_sys v0.1.0 (https://github.com/ngaut/rust-rocksdb.git#e2a50c47)`
process didn't exit successfully: `/home/suzhou/zhousu/tidb/docs/scripts/deps/tikv/target/release/build/librocksdb_sys-10e3c64f88ec5fe0/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-unknown-linux-gnu")
OPT_LEVEL = Some("3")
PROFILE = Some("release")
TARGET = Some("x86_64-unknown-linux-gnu")
debug=false opt-level=3
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CXX_x86_64-unknown-linux-gnu = None
CXX_x86_64_unknown_linux_gnu = None
HOST_CXX = None
CXX = None
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-unknown-linux-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CXXFLAGS_x86_64-unknown-linux-gnu = None
CXXFLAGS_x86_64_unknown_linux_gnu = None
HOST_CXXFLAGS = None
CXXFLAGS = None
running: "c++" "-O3" "-ffunction-sections" "-fdata-sections" "-m64" "-fPIC" "-std=c++11" "-fPIC" "-O2" "-o" "/home/suzhou/zhousu/tidb/docs/scripts/deps/tikv/target/release/build/librocksdb_sys-9b6727fa259b776b/out/crocksdb/c.o" "-c" "crocksdb/c.cc"
cargo:warning=crocksdb/c.cc:52:16: error: ‘rocksdb::IngestExternalFileOptions’ has not been declared
cargo:warning= using rocksdb::IngestExternalFileOptions;
cargo:warning=                ^
cargo:warning=crocksdb/c.cc:116:48: error: ‘IngestExternalFileOptions’ does not name a type
cargo:warning= struct crocksdb_ingestexternalfileoptions_t  { IngestExternalFileOptions rep; };
cargo:warning=                                                ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_iter_seek_for_prev(crocksdb_iterator_t*, const char*, size_t)’:
cargo:warning=crocksdb/c.cc:1054:14: error: ‘class rocksdb::Iterator’ has no member named ‘SeekForPrev’
cargo:warning=   iter->rep->SeekForPrev(Slice(k, klen));
cargo:warning=              ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingestexternalfileoptions_set_move_files(crocksdb_ingestexternalfileoptions_t*, unsigned char)’:
cargo:warning=crocksdb/c.cc:2388:8: error: ‘crocksdb_ingestexternalfileoptions_t {aka struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   opt->rep.move_files = move_files;
cargo:warning=        ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingestexternalfileoptions_set_snapshot_consistency(crocksdb_ingestexternalfileoptions_t*, unsigned char)’:
cargo:warning=crocksdb/c.cc:2394:8: error: ‘crocksdb_ingestexternalfileoptions_t {aka struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   opt->rep.snapshot_consistency = snapshot_consistency;
cargo:warning=        ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingestexternalfileoptions_set_allow_global_seqno(crocksdb_ingestexternalfileoptions_t*, unsigned char)’:
cargo:warning=crocksdb/c.cc:2400:8: error: ‘crocksdb_ingestexternalfileoptions_t {aka struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   opt->rep.allow_global_seqno = allow_global_seqno;
cargo:warning=        ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingestexternalfileoptions_set_allow_blocking_flush(crocksdb_ingestexternalfileoptions_t*, unsigned char)’:
cargo:warning=crocksdb/c.cc:2406:8: error: ‘crocksdb_ingestexternalfileoptions_t {aka struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   opt->rep.allow_blocking_flush = allow_blocking_flush;
cargo:warning=        ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingest_external_file(crocksdb_t*, const char* const*, size_t, const crocksdb_ingestexternalfileoptions_t*, char**)’:
cargo:warning=crocksdb/c.cc:2421:30: error: ‘class rocksdb::DB’ has no member named ‘IngestExternalFile’
cargo:warning=   SaveError(errptr, db->rep->IngestExternalFile(files, opt->rep));
cargo:warning=                              ^
cargo:warning=crocksdb/c.cc:2421:61: error: ‘const crocksdb_ingestexternalfileoptions_t {aka const struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   SaveError(errptr, db->rep->IngestExternalFile(files, opt->rep));
cargo:warning=                                                             ^
cargo:warning=crocksdb/c.cc: In function ‘void crocksdb_ingest_external_file_cf(crocksdb_t*, crocksdb_column_family_handle_t*, const char* const*, size_t, const crocksdb_ingestexternalfileoptions_t*, char**)’:
cargo:warning=crocksdb/c.cc:2432:30: error: ‘class rocksdb::DB’ has no member named ‘IngestExternalFile’
cargo:warning=   SaveError(errptr, db->rep->IngestExternalFile(handle->rep, files, opt->rep));
cargo:warning=                              ^
cargo:warning=crocksdb/c.cc:2432:74: error: ‘const crocksdb_ingestexternalfileoptions_t {aka const struct crocksdb_ingestexternalfileoptions_t}’ has no member named ‘rep’
cargo:warning=   SaveError(errptr, db->rep->IngestExternalFile(handle->rep, files, opt->rep));
cargo:warning=                                                                          ^
ExitStatus(ExitStatus(256))


command did not execute successfully, got: exit code: 1



--- stderr
thread 'main' panicked at 'explicit panic', /home/suzhou/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.35/src/lib.rs:897
note: Run with `RUST_BACKTRACE=1` for a backtrace.

Makefile:32: recipe for target 'release' failed
make: *** [release] Error 101
```

--- a/dev-guide/development.md
+++ b/dev-guide/development.md
@@ -8,7 +8,7 @@ Before you begin, check the [supported platforms](./requirements.md#supported-pl
 
 ## Install RocksDB
 
-RocksDB 4.12+ is required, You can install the RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) or use the [build RocksDB script](../scripts/build_rocksdb.sh) to build and install RocksDB in the system path.
+RocksDB 5.0+ is required, You can install the RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) or use the [build RocksDB script](../scripts/build_rocksdb.sh) to build and install RocksDB in the system path.
 
 ## Build TiKV
 

--- a/dev-guide/development.md
+++ b/dev-guide/development.md
@@ -8,7 +8,7 @@ Before you begin, check the [supported platforms](./requirements.md#supported-pl
 
 ## Install RocksDB
 
-RocksDB 5.0+ is required, You can install the RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) or use the [build RocksDB script](../scripts/build_rocksdb.sh) to build and install RocksDB in the system path.
+RocksDB 5.1 is required, You can install the RocksDB shared library manually according to [INSTALL.md](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) or use the [build RocksDB script](../scripts/build_rocksdb.sh) to build and install RocksDB in the system path.
 
 ## Build TiKV
 

--- a/scripts/build_rocksdb.sh
+++ b/scripts/build_rocksdb.sh
@@ -10,7 +10,7 @@ echo "building RocksDB in $DEPS_PATH"
 mkdir -p ${DEPS_PATH}
 cd $DEPS_PATH
 
-ROCKSDB_VER=4.12.fb
+ROCKSDB_VER=5.0.fb
 
 SUDO=
 if which sudo; then 

--- a/scripts/build_rocksdb.sh
+++ b/scripts/build_rocksdb.sh
@@ -10,7 +10,7 @@ echo "building RocksDB in $DEPS_PATH"
 mkdir -p ${DEPS_PATH}
 cd $DEPS_PATH
 
-ROCKSDB_VER=5.0.fb
+ROCKSDB_VER=v5.1.2
 
 SUDO=
 if which sudo; then 

